### PR TITLE
Add parallel Pippenger's on CPU via rayon

### PR DESCRIPTION
**Pull request by MegaRedHand**
# Add parallel Pippenger's on CPU via _rayon_

Depends on #326

## Description

This PR adds a parallel version of Pippenger's algorithm with support for rayon's iterator-based parallelization.

## Type of change

Please delete options that are not relevant.

- [X] Optimization

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [X] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [X] This change is an Optimization
  - [X] Benchmarks added/run

### CODE REVIEW

File: math/src/msm/pippenger.rs [hunk 1]
```diff
@@ -86,10 +94,67 @@ where
...
         .unwrap_or_else(G::neutral_element)
 }
 
+#[cfg(feature = "rayon")]
+// It has the following differences with the sequential one:
+//  1. It uses one vec per thread to store buckets.
+//  2. It reduces all window results via a different method.
+pub fn parallel_msm_with<const NUM_LIMBS: usize, G>(
+    cs: &[UnsignedInteger<NUM_LIMBS>],
+    hidings: &[G],
```
Comment (schouhy) [inline math/src/msm/pippenger.rs:108]: Could we name the second parameter `points` or something like that? In this abstract math context the term "hiding" doesn't really have a meaning.

**Review by schouhy — APPROVED** [review]
Awesome! Thanks! Left a minor comment

File: math/src/msm/pippenger.rs (hunk 1 above)
Comment (MegaRedHand) [reply math/src/msm/pippenger.rs:108]: Sure. I was just following the convention on the original `msm` function. I'll change it in all of them then.

Comment (MegaRedHand) [reply math/src/msm/pippenger.rs]: Done

**Review by pablodeymo — APPROVED** [review]
LGTM

**Review by Oppen — APPROVED** [review]

**Pull request merged**